packages/cli/commands/functions/deploy.js

```diff
@@ -68,21 +62,22 @@ exports.handler = async options => {
   );
 
   try {
-    spinner = makeSpinner(
-      'Building and deploying',
-      functionPath,
-      accountId
-    ).start();
+    spinnies = new Spinnies();
+    spinnies.add('buildAndDeployStatus', {
+      text: `Building and deploying bundle for '${functionPath}' on account '${accountId}'.\n`,
+    });
     const buildId = await buildPackage(accountId, functionPath);
     const successResp = await pollBuildStatus(accountId, buildId);
     const buildTimeSeconds = (successResp.buildTime / 1000).toFixed(2);
-    spinner.stop();
+    spinnies.remove('buildAndDeployStatus');
+    spinnies.stopAll();
     await outputBuildLog(successResp.cdnUrl);
     logger.success(
       `Built and deployed bundle from package.json for ${functionPath} on account ${accountId} in ${buildTimeSeconds}s.`
     );
   } catch (e) {
-    spinner && spinner.stop && spinner.stop();
+    spinnies && spinnies.remove && spinnies.remove('buildAndDeployStatus');
+    spinnies && spinnies.stopAll && spinnies.stopAll();
```

Comment:
Are these checks necessary (I assume you were just copying the existing pattern)? We might be able to remove them.

Reply:
Initializing the spinnies outside the try...catch block makes the checks unnecessary, thanks for the callout 👍 

A note: I was initially trying to remove the `spinnies` in a ...finally block, but that doesn't remove the spinnies in time for the error message in the catch block to correctly display. Ergo, we need to remove and stop spinnies in both blocks.